### ISSUE TITLE
tenantDomain infinite redirect

### ISSUE THREAD

**Issue by andy3471** [laravel-filament/filament#11896]
### Package

filament/filament

### Package Version

3.2.0

### Laravel Version

v11.0.7

### PHP Version

8.3

### Problem description

I am trying create an app which has three panels:

The Front Panel, which is tenanted
The Admin Panel, also tenanted,
A Super Admin panel, not tenanted

I want the front panel to use the tenants domain name, read from the database. According to the documentation, I can do this with:
```php          
 ->tenantDomain('{tenant:domain}')
```

I've added this to my front panel and admin panel provider, and also added the domain column to the tenants table. However, if I go to site-1.example.local (added to my hosts file), the login displays, however as soon as I log in, I get stuck in an infinite redirect loop back to site-1.example.local

I've noticed Filament has this in the routes file
```php
if ($hasTenancy) {
    Route::get('/', RedirectToTenantController::class)->name('tenant');
}
```

This seems to be the problem, as it repeatedly hits this controller method, which does the following:
```php
$panel = Filament::getCurrentPanel();
$tenant = Filament::getUserDefaultTenant(Filament::auth()->user());

if (! $tenant) {
    return $this->redirectToTenantRegistration($panel);
}

$url = $panel->getUrl($tenant);

if (blank($url)) {
    abort(404);
}

return redirect($url);
```

So it looks like on login, it redirects to the '/' route, which then repeatedly redirect back to the tenant. Am I missing some config or is this a bug?

Thanks in advance for the help!

### Expected behavior

Each tenant would work with their own domain. Going to that domain would show the filament login page, and signing in would take you straight to that tenant.

### Steps to reproduce

Pull the example repository
Copy the .env.example to .env
Run the database migrations + seed

Add the following you your hosts file for testing:

127.0.0.1       site-1.inform-ehr.local
127.0.0.1       site-2.inform-ehr.local
127.0.0.1       site-3.inform-ehr.local

Go to site-1.inform-ehr.local and try to login with test@example.com / password

You will be infinitely redirected.

### Reproduction repository

https://github.com/andy3471/filament-tenant-domain

**Comment by craigvc** [comment]
I have same issue :)

**Comment by craigvc** [comment]
issue seems to be line 27 in RedirectToTenantController @andy3471 I put panel name after url and it works (hacky)

**Comment by danharrin** [comment]
Fixed in #11940, please read the docs changes as there are tweaks to how slug params work & the addition of a route parameter.
